rls/src/main/java/io/grpc/rls/LinkedHashLruCache.java

```diff
@@ -243,27 +214,25 @@ public final List<V> values() {
    */
   protected final boolean fitToLimit() {
     boolean removedAnyUnexpired = false;
-    synchronized (lock) {
-      if (estimatedSizeBytes.get() <= estimatedMaxSizeBytes) {
-        // new size is larger no need to do cleanup
-        return false;
-      }
-      // cleanup expired entries
-      long now = ticker.read();
-      cleanupExpiredEntries(now);
-
-      // cleanup eldest entry until new size limit
-      Iterator<Map.Entry<K, SizedValue>> lruIter = delegate.entrySet().iterator();
-      while (lruIter.hasNext() && estimatedMaxSizeBytes < this.estimatedSizeBytes.get()) {
-        Map.Entry<K, SizedValue> entry = lruIter.next();
-        if (!shouldInvalidateEldestEntry(entry.getKey(), entry.getValue().value, now)) {
-          break; // Violates some constraint like minimum age so stop our cleanup
-        }
-        lruIter.remove();
-        // eviction listener will update the estimatedSizeBytes
-        evictionListener.onEviction(entry.getKey(), entry.getValue(), EvictionType.SIZE);
-        removedAnyUnexpired = true;
+    if (estimatedSizeBytes <= estimatedMaxSizeBytes) {
+      // new size is larger no need to do cleanup
```

Comment:
The reason given in the comment only describes what happens when it is called from this class, not from RlsAsyncLruCache.  The comment should probably just be removed as obvious, but could be changed to `there is space available so no need to do cleanup`

Reply:
Do you want to file an issue to track that?

Reply:
Not worth an issue, thought that you could do trivial cleanup even if not related to this change.